libcodechecker/libhandlers/parse.py

```diff
@@ -188,7 +188,55 @@ def parse(f, context, metadata_dict, suppress_handler, skip_handler, steps):
         analyzed_source_file = \
             metadata_dict['result_source_files'][f]
 
-    return rh.parse_and_write(f, analyzed_source_file)
+    files, reports = rh.parse(f)
+
+    plist_mtime = util.get_last_mod_time(f)
+
+    changed_files = False
+    for source_file in files:
+        file_mtime = util.get_last_mod_time(source_file)
+        if file_mtime > plist_mtime:
+            changed_files = True
+            LOG.warning(source_file +
+                        ' did change since the last analysis.')
+
+    if changed_files:
+        return {}, changed_files
+    else:
+        report_stats = rh.write(files, reports, analyzed_source_file)
+        return report_stats, changed_files
+
+
+def has_file_change(files, analyze_file_content_hashes):
```

Comment:
Do we still need this function and the one below?

Reply:
I've cleaned it up.